### PR TITLE
Fix the doc

### PR DESCRIPTION
**Pull request by jfsmig**
Many misleading information.

### CODE REVIEW

File: gridinit.conf [hunk 1]
```diff
@@ -43,59 +47,37 @@ limit.core_size=-1
...
+env.PATH=/bin:/usr/bin:/usr/local/bin
+env.LD_LIBRARY_PATH=/lib:/usr/lib:/usr/local/lib
+start_at_boot=true
 
-# start_at_boot : BOOL
-# Optional, default: true
-# Tells if the service has to be started when it is met for the first
-# time (i.e. not changed if the service is refreshed after a configuration
-# reload). 'start_at_boot' is overriden by 'enable' or CLI 'start' and
-# 'stop' commands.
```
Comment (fvennetier) [inline gridinit.conf:99]: This comment should stay somewhere! Giving a working configuration example is not enough.

**Review by fvennetier — COMMENTED** [review]
OK for the example.
Not OK for getting rid of the documentation!

**Comment by irevoire** [comment]
I was trying to compile this project and I noticed that `libtool` was not specified in the dependencies but was needed to build `libdill`.

I think we could add this in this PR instead of doing a new one.

**Comment by irevoire** [comment]
Ok so I was noticing a strange thing that was not happening on the master branch.
It look like the daemon is dying after the first call of `gridinit_cmd`.
I did a full install from scratch so you can see my exact setup:
```
$ pkg install gmake cmake gcc libevent glib libtool pkgconf autoconf automake git # all the deps for freeBSD
...
$ git clone https://github.com/open-io/gridinit
...
$ cd gridinit
$ git pull https://github.com/jfsmig/gridinit.git -- M-doc-20200120
remote: Enumerating objects: 10, done.
remote: Counting objects: 100% (10/10), done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 10 (delta 7), reused 10 (delta 7), pack-reused 0
Unpacking objects: 100% (10/10), done.
From https://github.com/jfsmig/gridinit
 * branch            M-doc-20200120 -> FETCH_HEAD
Updating d8104ec..b10f6e7
Fast-forward
 README.md       | 20 ++++++++++++++++++++
 gridinit.conf   | 78 ++++++++++++++++++++++++++++++------------------------------------------------
 main/gridinit.c |  2 +-
 3 files changed, 51 insertions(+), 49 deletions(-)
$ cmake -D CMAKE_C_COMPILER=gcc -D GRIDINIT_SOCK_PATH=/tmp/gridinit.sock .
...
$ make -j4
...
$ ./gridinit -d ./gridinit.conf
1579535175.872 71296 2C2A log INF gridinit reconfigure : loading main parameters from section [Default]
1579535175.907 71296 2C2A log INF gridinit reconfigure : managing service section [service.key-down]
1579535175.907 71296 2C2A log INF gridinit reconfigure : managing service section [service.key-disabled]
1579535175.907 71296 2C2A log INF gridinit reconfigure : managing service section [service.key1]
1579535175.908 71296 2C2A log NOT gridinit Service file pattern matched no file!
1579535175.908 71296 2C2A log INF gridinit Configuration loaded from [./gridinit.conf]
$
$ ./gridinit_cmd status
KEY  STATUS      PID GROUP
$
$ ./gridinit_cmd status
Connection to UNIX socket [/tmp/gridinit.sock] failed : Connection refused
$
$ ps aux
USER   PID %CPU %MEM  VSZ  RSS TT  STAT STARTED    TIME COMMAND
root 56179  0.0  0.0 6412 1856  -  SsJ  16:19   0:00.01 /usr/sbin/syslogd -ss
root 56235  0.0  0.0 6464 1848  -  IsJ  16:19   0:00.01 /usr/sbin/cron -J 60 -s
root 56261  0.0  0.0 6952 2160  0  IJ   16:19   0:00.01 login [pam] (login)
root 56265  0.0  0.0 7412 2984  0  SJ   16:19   0:00.12 -csh (csh)
root 71309  0.0  0.0 6912 2352  0  R+J  16:47   0:00.00 ps aux
```

**Comment by irevoire** [comment]
Here is another view of my problem:
In my first terminal I run:
```
130 irevoire@ovh ~/perso/M-doc-20200120 🌷 % ./gridinit  ./gridinit.conf                                                                                     [M-doc-20200120]:(
1579624100.840 41264 2C2A log INF gridinit reconfigure : loading main parameters from section [Default]
1579624100.841 41264 2C2A log INF gridinit reconfigure : managing service section [service.key-down]
1579624100.841 41264 2C2A log INF gridinit reconfigure : managing service section [service.key-disabled]
1579624100.841 41264 2C2A log INF gridinit reconfigure : managing service section [service.key1]
1579624100.841 41264 2C2A log NOT gridinit Service file pattern matched no file!
1579624100.841 41264 2C2A log INF gridinit Configuration loaded from [./gridinit.conf]
```
Notice I did not use the `-d` option so this time we’ll be able to see the errors.

On a second terminal:
```
irevoire@ovh ~/perso/M-doc-20200120 🐄 % ./gridinit_cmd status                                                                                                 [M-doc-20200120]
KEY  STATUS      PID GROUP
```

And now on the first terminal I get:
```
1579624100.842 41264 2C2A log INF gridinit Starting service [key1] with pid 41265
[1]    41264 segmentation fault (core dumped)  ./gridinit ./gridinit.conf
```

**Comment by jfsmig** [comment]
@irevoire This has been reproduced. A fix is on the way, it sems related to the usage of trampolines for nested functions used as callbacks (function pointers). It supersedes #42 by the way (since the fix gets rid of any nested func).

**Review by fvennetier — APPROVED** [review]

**Pull request merged**